src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,270 @@
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
+
+    /** Special time value to indicate the absence of an oldest trigger. */
+    private final static long NO_TRIGGER = 0;
+
+    /** The executor of this background job. */
+    private final Executor executor;
+    /** The ticker used to schedule this job. */
+    private final Ticker ticker;
+    /** Job wrapper for status tracking. */
+    private final DelayedBackgroundRunnable realJob;
+    /** Human-readable name of this job. */
+    private final String name;
+    /** Aggregation delay in milliseconds. */
+    private final long delay;
+
+    /** The time at which the first trigger occurred after the latest job execution start. */
+    private final AtomicLong oldestTrigger = new AtomicLong(NO_TRIGGER);
+    /** Running state of this job. */
+    private JobState state = JobState.IDLE;
+    /** Job execution thread, only if the job is running. */
+    private Thread thread = null;
+
+    /**
+     * Constructs a delayed background job.
+     * The {@link Executor} and {@link Ticker} given <b>must</b> have an asynchronous implementation
+     * of respectively {@link Executor#execute(Runnable, String) execute} and
+     * {@link Ticker#queueTimedJob(Runnable, String, long, boolean, boolean) queueTimedJob}.
+     * @param job the job to run in the background
+     * @param name a human-readable name for the job
+     * @param delay the background job aggregation delay in milliseconds
+     * @param executor an asynchronous executor
+     * @param ticker an asynchronous ticker
+     */
+    DelayedBackgroundJob(Runnable job, String name, long delay, Executor executor, Ticker ticker) {
+        this.executor = executor;
+        this.ticker = ticker;
+        this.name = name;
+        this.delay = delay;
+        this.realJob = new DelayedBackgroundRunnable(job);
+    }
+
+    /**
+     * Triggers scheduling of the job, if no job is scheduled. All subsequent triggers are ignored
+     * until the job executes.
+     *
+     * The first trigger received after the start of the last job execution leads to scheduling of
+     * another execution of the job, either after the delay or when the currently executing job is
+     * finished, whichever comes last. A newly constructed delayed background job can be assumed to
+     * have started its last job infinitely in the past.
+     */
+    @Override
+    public void trigger() {
+        if (oldestTrigger.compareAndSet(NO_TRIGGER, System.currentTimeMillis())) {
+            tryEnqueue();
+        }
+    }
+
+    /**
+     * If this job is {@code IDLE}, enqueue the next run in {@link #delay} milliseconds after
```

Comment:
To speed up the release, I've fixed this myself in commit 7ebeea440a59bee25fa45f66a8e10be33fe55ffe.

-> Done.